### ISSUE TITLE
Centralize Serializers

### ISSUE THREAD

**Issue by vaage** [vaage/Chat-Service-Sandbox#11]
I do not like how serialization code is scattered through out the whole code base. We should move all serialization code into a centralize section of the code and decouple serialization from the implementation of the types.

**Comment by vaage** [comment]
To help make this easier to work with create a Reader and Writer class to handle all reading and writing of types to an output stream:

```
  final InputStream in = ...

  final ChatReader reader = new ChatReader(in);

  final Uuid message = reader.readUuid();
  final Uuid user = reader.readUuid();
  final Uuid conversation = reader.readUuid();
  final String body = reader.readString();
  final String time = reader.readTime();

  return new Message(message, user, conversation, body, time);
```

**Comment by vaage** [comment]
This was closed with #26
